harness/determined/_execution.py

```diff
@@ -6,10 +6,25 @@
 from typing import Any, Dict, Iterator, List, Optional, Tuple
 
 import determined as det
-from determined import constants, gpu, horovod, workload
+from determined import constants, gpu, horovod, util, workload
 from determined_common import api
 
 
+class InvalidHP(Exception):
+    """Raised by user when InvalidHP exception is encountered
+
+    Attributes:
+        message -- explanation of the error
+    """
+
+    def __init__(self, message: str = "Invalid hyperparameter exception encountered") -> None:
+        self.message = message
+        super().__init__(self.message)
+
+    def __str__(self) -> str:
+        return repr(self.message)
+
+
```

Comment:
I actually think that this would probably suffice:
```python
class InvalidHP(Exception):
    def __init__(self, msg: str = "...") -> None:
        if not isinstance(msg, str):
            raise TypeError("InvalidHP exceptions can be initialized with a custom message, "
                f"but it must be a string, not {type(msg).__name__}")
        self.msg = msg
```
I don't think the `__str__` is needed or the super().__init__.